### PR TITLE
fix(oui-pagination): add input search when huge amount of pages

### PR DESCRIPTION
**Pull request by AxelPeter**
Close https://github.com/ovh-ux/ovh-ui-angular/issues/354

**Related to**
- https://github.com/ovh-ux/ovh-ui-kit/pull/342
- https://github.com/ovh-ux/ovh-ui-kit-documentation/pull/183

### CODE REVIEW

File: packages/oui-pagination/src/pagination.controller.js [hunk 1]
```diff
@@ -35,17 +37,24 @@ export default class {
...
     processPaginationChange () {
         this.pageCount = this.getPageCount();
-        this.pageRange = this.getPageRange();
         this.currentPage = this.getCurrentPage();
+        this.inputPage = this.getCurrentPage(); // Update model for input
     }
 
     processTranslations () {
-        this.translations = Object.assign({}, this.config.translations);
+        const translations = Object.assign({}, this.config.translations);
```
Comment (FredericEspiau) [inline packages/oui-pagination/src/pagination.controller.js]:
```suggestion
        const translations = {...this.config.translations};
```

File: packages/oui-pagination/src/pagination.controller.js [hunk 2]
```diff
@@ -35,17 +37,24 @@ export default class {
...
+
+        this.translations = angular.copy(translations);
```
Comment (FredericEspiau) [inline packages/oui-pagination/src/pagination.controller.js]:
```suggestion
        this.translations = {...translations};
```

File: packages/oui-pagination/src/pagination.controller.js [hunk 3]
```diff
@@ -71,9 +80,19 @@ export default class {
 
     onPageChange (page) {
         this.currentOffset = (this.pageSize * (page - 1)) + 1;
+        this.currentPage = this.getCurrentPage();
+        this.inputPage = this.getCurrentPage(); // Update model for input
         this._onChange();
     }
 
+    checkPageRange (page) {
+        const currentPage = typeof page === "number" ?
```
Comment (FredericEspiau) [inline packages/oui-pagination/src/pagination.controller.js]:
```suggestion
        const currentPage = Number.isInteger(page) ?
```

File: packages/oui-pagination/src/pagination.controller.js [hunk 4]
```diff
@@ -71,9 +80,19 @@ export default class {
...
+            Math.min(Math.max(page, 1), this.pageCount) :
```
Comment (FredericEspiau) [inline packages/oui-pagination/src/pagination.controller.js]: `_.clamp` ? Not sure I understand

File: packages/oui-pagination/src/pagination.html [hunk 5]
```diff
@@ -47,30 +49,54 @@
...
+
+        <!-- Numeric (> 100 pages) -->
+        <form class="oui-pagination-numeric"
+            ng-if="$ctrl.pageCount > 100"
+            ng-submit="$ctrl.checkPageRange($ctrl.inputPage)"
+            novalidate>
+            <span aria-hidden="true"
+                ng-bind="$ctrl.translations.inputPageOfPageCount[0]">
+            </span>
+            <input class="oui-pagination-numeric__input oui-input oui-input_xs"
```
Comment (FredericEspiau) [inline packages/oui-pagination/src/pagination.html]: No `label` ?

Comment (AxelPeter) [reply packages/oui-pagination/src/pagination.html]: No need here. It render something like:

`Page [input] of x`

It's the reason I put the aria-hidden on the `span`, and a full label on the input.

File: packages/oui-pagination/src/pagination.controller.js (hunk 3 above)
Comment (AxelPeter) [reply packages/oui-pagination/src/pagination.controller.js]: Seems better to test for integer :)

File: packages/oui-pagination/src/pagination.controller.js (hunk 1 above)
Comment (AxelPeter) [reply packages/oui-pagination/src/pagination.controller.js]: Yes, I prefer this implementation.

File: packages/oui-pagination/src/pagination.controller.js (hunk 4 above)
Comment (FredericEspiau) [reply packages/oui-pagination/src/pagination.controller.js]: According to @jleveugle yes it is, it's yours to see ¯\_(ツ)_/¯

File: packages/oui-pagination/src/pagination.html (hunk 5 above)
Comment (FredericEspiau) [reply packages/oui-pagination/src/pagination.html]: Seems like a nice pattern

File: packages/oui-pagination/src/pagination.controller.js (hunk 4 above)
Comment (AxelPeter) [reply packages/oui-pagination/src/pagination.controller.js]: If it's for @jleveugle, let's do it !

**Pull request merged**